crates/sqlbuiltins/src/functions/table/parquet_metadata.rs

```diff
@@ -0,0 +1,283 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use datafusion::arrow::array::{Int64Array, RecordBatch, StringArray};
+use datafusion::arrow::datatypes::{DataType, Field, Schema, SchemaRef};
+use datafusion::datasource::TableProvider;
+use datafusion::error::Result;
+use datafusion::execution::context::SessionState;
+use datafusion::logical_expr::{Expr, Signature, Volatility};
+use datafusion::parquet::basic::ConvertedType;
+use datafusion::parquet::file::reader::{FileReader, SerializedFileReader};
+use datafusion::parquet::file::statistics::Statistics;
+use datafusion::physical_plan::memory::MemoryExec;
+use datafusion::physical_plan::ExecutionPlan;
+use datafusion_ext::functions::{FuncParamValue, TableFuncContextProvider};
+use datasources::common::url::DatasourceUrl;
+use protogen::metastore::types::catalog::{FunctionType, RuntimePreference};
+
+use super::TableFunc;
+use crate::functions::ConstBuiltinFunction;
+
+/// Modified implementation from datafusion-cli https://github.com/GlareDB/arrow-datafusion/blob/777235199414c7e83ffd5f5008fd2de8139a83d7/datafusion-cli/src/functions.rs#L296
+/// It's slightly adapted to work with our codebase
+/// Currently this only works with local files. We can eventually add support for remote files.
+pub struct ParquetMetadataFunc;
+
+#[async_trait]
+impl TableFunc for ParquetMetadataFunc {
+    fn detect_runtime(
+        &self,
+        _args: &[FuncParamValue],
+        _parent: RuntimePreference,
+    ) -> datafusion_ext::errors::Result<RuntimePreference> {
+        Ok(RuntimePreference::Local)
+    }
+
+    async fn create_provider(
+        &self,
+        _ctx: &dyn TableFuncContextProvider,
+        args: Vec<FuncParamValue>,
+        _opts: HashMap<String, FuncParamValue>,
+    ) -> datafusion_ext::errors::Result<Arc<dyn TableProvider>> {
+        let filename: DatasourceUrl = match args.first() {
```

Comment:
is it neccesary/desired to error if the function is called with more than one file? 

Is it useful to have this support a glob?

Reply:
I don't think we'd want to support globbing, IDK how we'd want to handle schema mismatches or other such errors. This was very much just a copy/paste of datafusion's implementation to get things started. 

I can add an assertion for multiple arguments, but don't want to try to get overly complex with this initial implementation.